Marketplace/voiceHUDApp/appResources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,836 @@
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* (Philip) Script that shows me who is talking by making a cylinder above their head that 
+animates with their voice intensity when it is above a threshold
+ - a good reference would be the "Loud App"*/
+/* global EventBridge Users AccountServices Agent Avatar SoundCache EntityViewer */
+
+(function(){
+
+    // *************************************
+    // START UTILITY FUNCTIONS
+    // *************************************
+    
+    var POWER_UP = SoundCache.getSound(Script.resolvePath('./resources/sounds/rezzing.wav'));    
+    var injector;
+    function playSound(sound) {
+        if (sound.downloaded) {
+            if (injector) {
+   
+                injector.stop();
+            }
+            injector = Audio.playSound(sound, {
+                position: MyAvatar.position,
+                volume: 0.2,
+                localOnly: true
+            });
+        } else {
+            console.log("no sound downloaded");
+        }
+    }
+    
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+    // This function loads appui and connects to the needed signals
+    var AppUi = Script.require('appUi');
+    var ui;
+    function startup() {
+        ui = new AppUi({
+            home: Script.resolvePath("./resources/voiceHUD_ui.html"),
+            buttonName: "VOICE-HUD", // The name of your app
+            graphicsDirectory: Script.resolvePath("./resources/images/"), // Where your button icons are located
+            onMessage: onMessage
+        });       
+        Script.scriptEnding.connect(scriptEnding);
+        Window.domainChanged.connect(onDomainChange);
+    }
+ 
+
+    // This function starts and stops the app, triggered by the tablet UI
+    var enabled = false;
+    function toggleApp() {
+        if (!enabled){
+            startUpdate();
+            addAll();
+            playSound(POWER_UP);
+            enabled = true; 
+        } else {
+            stopUpdate();
+            removeAll();
+            enabled = false;
+        }
+    }
+
+
+    // This function decides how to handle web events from the tablet UI.
+    function onMessage(data) {
+        // EventBridge message from HTML script.
+        switch (data.type) {
+            case "EVENT_BRIDGE_OPEN_MESSAGE":
+                ui.sendToHtml({
+                    type: "buttonStatus",
+                    value: enabled
+                });
+                break;
+            case "TOGGLE_APP":
+                toggleApp();
+                ui.sendToHtml({
+                    type: "buttonStatus",
+                    value: enabled
+                });
+                break;
+        }
+    }
+
+
+    // This function keeps the app running even if you change domains.
+    var DOMAIN_DELAY = 100;
+    function onDomainChange(){
+        // turn it off on domain change
```

Comment:
this comment says "turn it off", but it looks like you're not doing that. on purpose?

Reply:
Fixed, I figured out how to keep it on after a domain change.